### PR TITLE
chore: pass in file reader to `FileManager` at runtime

### PR DESCRIPTION
**Pull request by TomAFrench**
# Description

## Summary\*

We currently conditionally compile how the `FileManager` reads in files based on the target architecture. If we were to pass this in as a boxed closure we could remove the need for `noir-source-resolver` entirely. `wasm` would then have full control on how to load Noir source.

This will add some overhead to reading in files however so I'd be interested to hear any opinions one way or the other. I'm preferring this solution to a generic to avoid that spreading and making the whole compiler generic.

cc @phated @kevaundray 

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by kevaundray** [comment]
> This will add some overhead to reading in files however so I'd be interested to hear any opinions one way or the other.

What overhead are you referring to?

**Comment by TomAFrench** [comment]
>What overhead are you referring to?

Rather than just calling the function which is defined at runtime, it'll need to go through the extra layer of dealing with the `Box`.

File: compiler/fm/src/file_reader.rs [hunk 1]
```diff
@@ -33,7 +35,7 @@ cfg_if::cfg_if! {
 
         }
 
-        pub(crate) fn read_file_to_string(path_to_file: &Path) -> Result<String, Error> {
+        pub fn read_file_to_string(path_to_file: &Path) -> Result<String, Error> {
```
Comment (kevaundray) [inline compiler/fm/src/file_reader.rs:38]: Could we move the code in `if #[cfg(target_arch = "wasm32")] {` to compiler/wasm?

Comment (TomAFrench) [reply compiler/fm/src/file_reader.rs:38]: Yeah, that's the plan. I just wanted to sound out the idea before doing the work.

Comment (kevaundray) [reply compiler/fm/src/file_reader.rs:38]: I think moving the file system impurity more towards `compiler/wasm` looks okay

**Review by kevaundray — APPROVED** [review]

**Review by kevaundray — APPROVED** [review]

**Pull request merged**